docs/guides/evm/understanding_the_mining_process.rst

```diff
@@ -0,0 +1,181 @@
+                difficulty: int) -> None:
+      validate_length(mix_hash, 32, title="Mix Hash")
+      validate_length(mining_hash, 32, title="Mining Hash")
+      validate_length(nonce, 8, title="POW Nonce")
+      cache = get_cache(block_number)
+      mining_output = hashimoto_light(
+          block_number, cache, mining_hash, big_endian_to_int(nonce))
+      if mining_output[b'mix digest'] != mix_hash:
+          raise ValidationError("mix hash mismatch; {0} != {1}".format(
+              encode_hex(mining_output[b'mix digest']), encode_hex(mix_hash)))
+      result = big_endian_to_int(mining_output[b'result'])
+      validate_lte(result, 2**256 // difficulty, title="POW Difficulty")
+
+Just by looking at the signature of that function we can see that validating the PoW is based on the following parameters:
+
+* ``block_number`` - the number of the given block
+* ``mining_hash`` - fill with terse description
+* ``mix_hash`` - fill with terse description
+* ``nonce`` - fill with terse description
+* ``difficulty`` - the difficulty of the PoW algorithm
+
+
+Finding a valid block
+---------------------
+
+Looking at the exception and the implementation of :func:`~evm.consensus.pow.check_pow` we may be left wondering: Where do we set the ``mix_hash`` that the error is complaining about?
+
+We can pass a number of key-value pairs to :func:`~~evm.chains.base.Chain.mine_block` to write these values to the header of our block. Let's set ``mix_hash`` to the value we were told that was expected and try to mine the block again.
+
+::
+
+  pow_args = {
+      'mix_hash': decode_hex('0x7a76bbf0c8d0e683fafa2d7cab27f601e19f35e7ecad7e1abb064b6f8f08fe21'),
+  }
+
+  chain.mine_block(**pow_args)
+
+Shoot! We're running into another error, this time telling us that the ``difficulty`` isn't with the expected range.
+
+.. code-block:: console
+
+  Traceback (most recent call last):
+    File "scripts/benchmark/run.py", line 111, in <module>
+      run()
+    File "scripts/benchmark/run.py", line 52, in run
+      block = chain.mine_block(**pow_args)
+    File "/py-evm/evm/chains/base.py", line 545, in mine_block
+      self.validate_block(mined_block)
+    File "/py-evm/evm/chains/base.py", line 585, in validate_block
+      self.validate_seal(block.header)
+    File "/py-evm/evm/chains/base.py", line 622, in validate_seal
+      header.mix_hash, header.nonce, header.difficulty)
+    File "/py-evm/evm/consensus/pow.py", line 72, in check_pow
+      validate_lte(result, 2**256 // difficulty, title="POW Difficulty")
+    File "/py-evm/evm/validation.py", line 92, in validate_lte
+      title=title,
+  evm.exceptions.ValidationError: POW Difficulty
+  50084831392019586920263880526078408578250402437811116223241331859679343306571
+  is not less than or equal to
+  882992383764307249142653314182893391999679604880738805815775866336575232
+
+Let's make another adjustment and also set the ``difficulty`` to the value we were told that was expected and try again.
```

Comment:
cc @carver 

@cburgdorf I know this is the path that you went down, but I'm skeptical of the value it provides to fully explore this in the docs.  It might be more valuable to provide a brief example of how someone *can* CPU mine a block if they so desire, and just to document what values are required to *properly* mine a POW block and potentially link out to external resources for details of the actual mining algorithm.

Reply:
Yep, I've been wondering the same thing as I wrote it and I agree that this is more an tutorial, "let's tinker with this" kind of approach.  Maybe we better follow the "happy path" and put in an info box that quickly explains why we can't just try to make up values like this.